### PR TITLE
CMR-5182 Add New CMR Dashboard ACL rules

### PR DESCRIPTION
**Pull request by daniel-zamora**
This PR Implements the following ACL targets and permissions
System Level: "Dashboard Admin Role" - CRUD, "Dashboard ARC Curator Role" - CRUD
Provider Level: "Dashboard DAAC Curator Role" - CRUD

### CODE REVIEW

**Comment by daniel-zamora** [comment]
I don't know that the test is 100% necessary in this case, but I added one anyway.

File: access-control-app/src/cmr/access_control/services/acl_validation.clj [hunk 1]
```diff
@@ -41,6 +41,7 @@
                               "CATALOG_ITEM_ACL"                [c r u d]
                               "INGEST_MANAGEMENT_ACL"           [r u]
                               "DATA_QUALITY_SUMMARY_DEFINITION" [c u d]
+                              "DASHBOARD_DAAC_CURATOR_ROLE"     [c r u d]
```
Comment (chris-durbin) [inline access-control-app/src/cmr/access_control/services/acl_validation.clj]: I really like this map.

Comment (daniel-zamora) [reply access-control-app/src/cmr/access_control/services/acl_validation.clj]: Yea, the guy who made it is really smart.

**Comment by oubiwann** [comment]
image

**Comment by oubiwann** [comment]
Okay, I fixed the PR. Updating to approved ...

**Comment by oubiwann** [comment]
Oh, SWEET gif update -- well done, SIR!

File: access-control-app/int-test/cmr/access_control/int_test/acl_crud_test.clj [hunk 2]
```diff
@@ -1164,4 +1163,75 @@
...
+    (is (= {"DASHBOARD_ADMIN" ["read" "create" "update" "delete"]}
+           (json/parse-string
+             (ac/get-permissions (u/conn-context) {:user_id "user1"
+                                                   :system_object "DASHBOARD_ADMIN"}
+                                 {:token token}))))
+    (is (= {"DASHBOARD_ARC_CURATOR" ["read" "create" "update" "delete"]}
+           (json/parse-string
+             (ac/get-permissions (u/conn-context) {:user_id "user1"
+                                                   :system_object "DASHBOARD_ARC_CURATOR"}
+                                 {:token token}))))))
```
Comment (sxu123) [inline access-control-app/int-test/cmr/access_control/int_test/acl_crud_test.clj]: Could you add a test that covers a user that is in a group that doesn't have all the permissions? Also, what would happen if say DASHBOARD_ARC_CURATOR allows only "read", and you have {:group_id group1-concept-id :permissions ["read" "create" "delete" "update"]}? I'm still not clear about which one determines the final permission.

Comment (daniel-zamora) [reply access-control-app/int-test/cmr/access_control/int_test/acl_crud_test.clj]: group permissions is an array value, meaning there can be multiple sets of permissions for different groups/user-types.

Comment (daniel-zamora) [reply access-control-app/int-test/cmr/access_control/int_test/acl_crud_test.clj]: The get-permissions allow for you to specify which user you want to check, and that user is a part of group1, so group1's permissions apply

Comment (daniel-zamora) [reply access-control-app/int-test/cmr/access_control/int_test/acl_crud_test.clj]: A user gets the super set of all permissions available to any group they are a part of, so if a user is in two groups, it gets both.  So if 1 group is read create, the other is update delete, they get create read update delete returned.

**Pull request merged**